### PR TITLE
tests/ci: simplify CI setup script

### PR DESCRIPTION
**Pull request by adrianreber**
The CI setup script still contained a couple of if 8 or 9 code paths which have been removed. This change also replaces of common used URLs and paths with variables.

Once we have a release repository this script needs to be adapted once more to use the release repository in combination with the latest factory project on OBS.

### CODE REVIEW

File: tests/ci/prepare-ci-environment.sh [hunk 1]
```diff
@@ -44,51 +47,34 @@ loop_command() {
...
+loop_command "${PKG_MANAGER}" "${YES}" install "${COMMON_PKGS}"
 if [ "${PKG_MANAGER}" = "dnf" ]; then
+	# If there is a release, this is the place to install the corresponding
+	# release RPM. As long as there is no release RPM only the OBS repository
+	# is used directly.
 	if [ "${ID}" = "openEuler" ]; then
-		loop_command "${PKG_MANAGER}" -y  install "${COMMON_PKGS}"
-		FACTORY_VERSION=3.0
-		loop_command wget http://obs.openhpc.community:82/OpenHPC3:/"${FACTORY_VERSION}":/Factory/openEuler_22.03/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/yum.repos.d/ohpc-pre-release.repo
+		loop_command wget "${OBS_SERVER}""${PROJECT}"/"${FACTORY_VERSION}":/Factory/openEuler_22.03/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/yum.repos.d/ohpc-pre-release.repo
```
Comment (martin-g) [inline tests/ci/prepare-ci-environment.sh:66]: Is `/Factory/openEuler_22.03/OpenHPC3:` intentionally not using `"${PROJECT}"` ?

File: tests/ci/prepare-ci-environment.sh [hunk 2]
```diff
@@ -44,51 +47,34 @@ loop_command() {
...
-		if [ "$?" -eq "11" ]; then
-			OHPC_RELEASE="http://repos.openhpc.community/OpenHPC/2/CentOS_8/${UNAME_M}/ohpc-release-2-1.el8.${UNAME_M}.rpm"
-		else
-			FACTORY_VERSION=3.0
-			# This is our RHEL 9 pre-release repository
-			loop_command wget http://obs.openhpc.community:82/OpenHPC3:/"${FACTORY_VERSION}":/Factory/EL_9/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/yum.repos.d/ohpc-pre-release.repo
-			NINE=1
-		fi
-		set -e
+		loop_command wget "${OBS_SERVER}""${PROJECT}"/"${FACTORY_VERSION}":/Factory/EL_9/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/yum.repos.d/ohpc-pre-release.repo
```
Comment (martin-g) [inline tests/ci/prepare-ci-environment.sh:68]: same here

File: tests/ci/prepare-ci-environment.sh [hunk 3]
```diff
@@ -44,51 +47,34 @@ loop_command() {
...
 	fi
 else
-	FACTORY_VERSION=3.0
-	loop_command "${PKG_MANAGER}" -n  install "${COMMON_PKGS}"
-	loop_command wget http://obs.openhpc.community:82/OpenHPC3:/"${FACTORY_VERSION}":/Factory/Leap_15/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/zypp/repos.d/ohpc-pre-release.repo
+	loop_command wget "${OBS_SERVER}""${PROJECT}"/"${FACTORY_VERSION}":/Factory/Leap_15/OpenHPC3:"${FACTORY_VERSION}":Factory.repo -O /etc/zypp/repos.d/ohpc-pre-release.repo
```
Comment (martin-g) [inline tests/ci/prepare-ci-environment.sh:71]: OK! I guess it is intentional! :-)

Comment (adrianreber) [reply tests/ci/prepare-ci-environment.sh:71]: I just overlooked that.

Comment (martin-g) [reply tests/ci/prepare-ci-environment.sh:71]: I'll improve it!

**Pull request merged**